src/ILCompiler.Compiler/src/Compiler/ExportedMethodsRootProvider.cs

```diff
@@ -18,27 +21,41 @@ public ExportedMethodsRootProvider(EcmaModule module)
             _module = module;
         }
 
-        public void AddCompilationRoots(IRootingServiceProvider rootProvider)
+        public IEnumerable<EcmaMethod> ExportedMethods
         {
-            foreach (var type in _module.GetAllTypes())
+            get
             {
-                foreach (var method in type.GetMethods())
-                {
-                    EcmaMethod ecmaMethod = (EcmaMethod)method;
+                var methods = new List<EcmaMethod>();
```

Comment:
Building the List of methods made sense when it was cached. Without cache, it would be better to `yield return` the methods.

Either keep the cache, or replace List with `yield return`.